book/controller.rst

```diff
@@ -675,9 +674,9 @@ the ``notice`` message:
             </div>
         <?php endforeach ?>
 
-By design, flash messages are meant to live for exactly one request (they're
-"gone in a flash"). They're designed to be used across redirects exactly as
-you've done in this example.
+.. tip::
+
+    Please note that using get() method in session :class:`Symfony\\Component\\HttpFoundation\\Session\\Flash\\FlashBag` object, automatically remove messages from session. If you want preserve flash messages, consider use peek() method instead.
```

Comment:
I think we should keep the existing paragraph and just reword it a bit. We can then add a tip for the `peek()` method like this:

```rst
By design, flash messages are meant to be processed exactly once. This means
that they vanish from the session automatically when they are retrieved from
the flash bag by calling the ``get()`` method.

.. tip::

    You can use the
    :method:`Symfony\\Component\\HttpFoundation\\Session\\Flash\\FlashBagInterface::peek()`
    method instead to retrieve the message while keeping it in the bag.
```

What do you think?